syntax = "proto3";

package greet;
service Greeter {
  //unario
  rpc SayHello (HelloRequest) returns (HelloReply);

  //server streaming
  rpc ParrotSaysHello (HelloRequest) returns (stream HelloReply);

  //client streaming
  rpc ChattyClientSayHello (stream HelloRequest) returns (DelayedReply);

  //Ambos streaming
  rpc InteractingHello (stream HelloRequest) returns (stream HelloReply);
}

message HelloRequest{
    string name = 1;
    string greeting = 2;
}

message HelloReply {
    string message = 1;
}

message DelayedReply{
    string message = 1;
    repeated HelloRequest request = 2;
}